src/module.c

```diff
@@ -8577,9 +8577,11 @@ void processModuleLoadingProgressEvent(int is_aof) {
     }
 }
 
-/* When a module key is deleted (in dbAsyncDelete/dbSyncDelete/dbOverwrite), it 
+/* When a key is deleted (in dbAsyncDelete/dbSyncDelete/dbOverwrite), it 
 *  will be called to tell the module which key is about to be released. */
 void moduleNotifyKeyUnlink(robj *key, robj *val, int dbid) {
+    dictPauseRehashing(server.db[dbid].dict);
```

Comment:
ohh, i see you handled dict rehashing here... well, maybe it's better / safer for dict.c to do that implicitly?

Reply:
At first I think we may use `dictFindWithPlink` but not use `dictFreePlinkEntry`.
Or we should always use them  in pairs

Reply:
looking at name (`dictFindWithPlink`) it indeed doesn't imply that you must call the second method.
but do we see any reason anyone will ever wanna use one and not the other?
if we don't, maybe we can rename it (to `dictTwoPhaseUnlinkFind` and `dictTwoPhaseUnlinkFree`).
if we do, then maybe just add some bold comment about caller needing to call `dictPauseRehashing` and avoid any modifications to the dict between these calls)

Reply:
The last thing is how should we do with this. I'm not sure.

Reply:
What do yo mean? 
Isn't that what I mentioned here: https://github.com/redis/redis/pull/9406#discussion_r698377459
Plus the rename I mentioned above.

Reply:
> looking at name (`dictFindWithPlink`) it indeed doesn't imply that you must call the second method.
> but do we see any reason anyone will ever wanna use one and not the other?
> if we don't, maybe we can rename it (to `dictTwoPhaseUnlinkFind` and `dictTwoPhaseUnlinkFree`).
> if we do, then maybe just add some bold comment about caller needing to call `dictPauseRehashing` and avoid any modifications to the dict between these calls)

I mean this.

Reply:
I may be missing something. It's just a rename. 
I. E. 
1. Rename these two functions so that it's clear both must be called. 
2. Increment and decrement d->pauserehash in there. 
3. Remove the calls to dictPauseRehashing
4. Maybe improve or revise the doc comment

Reply:
changes LGTM